### ISSUE TITLE
The function getNextId in 02_Challenge and 02_Solution does not behave correctly

### ISSUE THREAD

**Issue by maerqin** [LinkedInLearning/typescript-EssT-2428199#3]
## Issue Overview
The function getNextId always return the number 1, independent of the array given. Shouldn't the function return the biggest id plus 1? I fixed the issue by switching max and x.id in the conditional.

## Describe your environment
Typescript Version 4.7.4
node.js v16.16.0

## Steps to Reproduce
1. Run the program.
2. The property id is 1 instead of 4 of the next todo item.

## Expected Behavior
The property id from the next todo item should be 4.

## Current Behavior
The id is 1.

## Possible Solution
Switch max and x.id in the conditional.

**Comment by echarbeto** [comment]
Yes. It should be: x.id > max ? x.id : max since we want it to always return the largest value.
`function getNextId(items:Array<item>): number {
    return items.reduce((max, x) => x.id > max ? x.id : max, 0) + 1;
}`
It can also be replaced by this instruction:
`function getNextId(items:Array<item>): number {
    return Math.max(...items.map(item => item.id)) + 1;
}`
since an accumulator is not needed but rather return an id with the maximum value.
